client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -122,10 +123,27 @@ private Request buildQueryRequest(ClientSession session, String query)
             throw new ClientException("Invalid server URL: " + session.getServer());
         }
         url = url.newBuilder().encodedPath("/v1/statement").build();
-
-        Request.Builder builder = prepareRequest(url)
-                .post(RequestBody.create(MEDIA_TYPE_TEXT, query));
-
+        Request.Builder builder;
+        if (session.getPreprocessorClass().isPresent()) {
+            try {
+                Class<QueryPreprocessor> preprocessorClazz = (Class<QueryPreprocessor>) Class.forName(
+                        session.getPreprocessorClass().get(),
+                        true,
+                        this.getClass().getClassLoader());
+
+                QueryPreprocessor preprocessor = preprocessorClazz.getDeclaredConstructor().newInstance();
```

Comment:
getDeclaredConstructor -> getConstructor

-- you don't want to try to access non-public ctor here